cardano-node/src/Cardano/CLI/Tx.hs

```diff
@@ -1,114 +1,234 @@
+import qualified Ouroboros.Consensus.Ledger.Byron as Byron
+import           Ouroboros.Consensus.Ledger.Byron (GenTx(..), ByronBlockOrEBB)
+import           Ouroboros.Consensus.Ledger.Byron.Config (ByronConfig)
+import qualified Ouroboros.Consensus.Protocol as Consensus
+
+import           Cardano.CLI.Ops
+import           Cardano.CLI.Tx.Submission
+import           Cardano.Common.Protocol
+import           Cardano.Config.Types (CardanoConfiguration(..))
+import           Cardano.Node.Configuration.Topology
+
+
+newtype TxFile =
+  TxFile FilePath
+  deriving (Eq, Ord, Show, IsString)
+
+newtype NewTxFile =
+  NewTxFile FilePath
+  deriving (Eq, Ord, Show, IsString)
 
 
-prettyAddress :: CC.Common.Address -> Text
+prettyAddress :: Common.Address -> Text
 prettyAddress addr = TL.toStrict
-  $  F.format CC.Common.addressF         addr <> "\n"
-  <> F.format CC.Common.addressDetailedF addr
+  $  F.format Common.addressF         addr <> "\n"
+  <> F.format Common.addressDetailedF addr
 
-txSpendUTxOByronPBFT
-  :: CC.Genesis.Config
-  -> SigningKey
-  -> NonEmpty TxIn
-  -> NonEmpty TxOut
-  -> GenTx (ByronBlockOrEBB ByronConfig)
-txSpendUTxOByronPBFT gc sk ins outs =
-  let txattrs  = mkAttributes ()
-      tx       = UnsafeTx ins outs txattrs
-      wit      = signTxId (configProtocolMagicId gc) sk (Crypto.hash tx)
-      ATxAux atx awit =
-        mkTxAux tx . V.fromList . take (NE.length ins) $ repeat wit
-  in mkByronTx $ ATxAux (reAnnotate atx) (reAnnotate awit)
+readByronTx :: TxFile -> IO (GenTx (ByronBlockOrEBB ByronConfig))
+readByronTx (TxFile fp) = do
+  txBS <- LB.readFile fp
+  case deserialiseOrFail txBS of
+    Left e -> throwIO $ TxDeserialisationFailed fp e
+    Right tx -> pure tx
 
-txSpendGenesisUTxOByronPBFT
-  :: CC.Genesis.Config
-  -> SigningKey
-  -> Address
-  -> NonEmpty TxOut
-  -> GenTx (ByronBlockOrEBB ByronConfig)
-txSpendGenesisUTxOByronPBFT gc sk genAddr outs =
-  let txattrs  = mkAttributes ()
-      tx       = UnsafeTx (pure txIn) outs txattrs
-      txIn    :: CC.UTxO.TxIn
-      txIn     = genesisUTxOTxIn gc sk genAddr
-      wit      = signTxId (configProtocolMagicId gc) sk (Crypto.hash tx) 
-      ATxAux atx awit = mkTxAux tx . V.fromList . pure $ wit
-  in mkByronTx $ ATxAux (reAnnotate atx) (reAnnotate awit)
-
-signTxId :: ProtocolMagicId -> SigningKey -> TxId -> CC.UTxO.TxInWitness
-signTxId pmid sk txid = CC.UTxO.VKWitness
+signTxId :: ProtocolMagicId -> SigningKey -> TxId -> UTxO.TxInWitness
+signTxId pmid sk txid = UTxO.VKWitness
   (Crypto.toVerification sk)
   (Crypto.sign
     pmid
     Crypto.SignTx
     sk
-    (CC.UTxO.TxSigData txid))
+    (UTxO.TxSigData txid))
 
 -- | Given a genesis, and a pair of signing key and address, reconstruct a TxIn
 --   corresponding to the genesis UTxO entry.
-genesisUTxOTxIn :: CC.Genesis.Config -> SigningKey -> Address -> CC.UTxO.TxIn
+genesisUTxOTxIn :: Genesis.Config -> SigningKey -> Common.Address -> UTxO.TxIn
```

Comment:
:+1:

Reply:
This might also be something to look at for refactoring.

Reply:
Done, no more `CC.` in cardano-node:
```
$ fgrep 'CC.' . -rI | wc -l
0
```

Reply:
Done, no more `CC.` in cardano-node:
```
[nix-shell:~/cardano-node]$ fgrep 'CC.' . -rI | wc -l
0
```

Reply:
Done!